superset/views/core.py

```diff
@@ -250,7 +254,7 @@ def override_role_permissions(self) -> FlaskResponse:
                     )
                     db_ds_names.add(fullname)
 
-        existing_datasources = ConnectorRegistry.get_all_datasources(db.session)
+        existing_datasources = DatasourceDAO.get_all_sqlatables_datasources(db.session)
```

Comment:
I feel like if we know that the class is sqlatable, it shouldn't go through the datasourceDao but rather the SqlaTable class or a SqlaTableDAO itself. I think the datasourceDao should only be for looking up objects that could be any one of the many new datasource types, but you don't specifically know which one it is.